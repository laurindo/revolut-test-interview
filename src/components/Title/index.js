import React from 'react';
import { Spring } from 'react-spring';
import PropTypes from 'prop-types';

const Title = ({ title }) => (
  <Spring from={{ opacity: 0 }} to={{ opacity: 1 }}>
    { props => ( <p style={ props }>{ title }</p>) }
  </Spring>
);

PropTypes.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Title;
