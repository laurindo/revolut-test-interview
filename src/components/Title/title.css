/* BUTTON */
/* primary */
/* secondary */
h2 {
  font-size: 2em;
  text-align: center; }
  h2.primary {
    color: #8c8d8e; }
  h2.secondary {
    color: #fff; }
