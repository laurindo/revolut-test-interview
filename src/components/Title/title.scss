@import '../../styles/scss/colors.scss';

h2 {
  font-size: 2em;
  text-align: center;

  &.primary {
    color: $gray;
  }

  &.secondary {
    color: $white;
  }
}
