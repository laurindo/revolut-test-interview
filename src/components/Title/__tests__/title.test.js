import React from 'react';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Title from '../index';

configure({ adapter: new Adapter() });

describe('Title Component', () => {
  test('should render the component', () => {
    const component = shallow(<Title title="Hello!!!" />);
    expect(component).toMatchSnapshot();
    expect(component.find('h2')).toBeDefined();
  });
});
