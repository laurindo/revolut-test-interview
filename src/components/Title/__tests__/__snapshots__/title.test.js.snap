exports[`Title Component should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    title="Hello!!!" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "config": Object {
        "friction": 26,
        "tension": 170,
      },
      "force": false,
      "from": Object {
        "opacity": 0,
      },
      "immediate": false,
      "inject": [Function],
      "native": false,
      "reset": false,
      "to": Object {
        "opacity": 1,
      },
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "config": Object {
          "friction": 26,
          "tension": 170,
        },
        "force": false,
        "from": Object {
          "opacity": 0,
        },
        "immediate": false,
        "inject": [Function],
        "native": false,
        "reset": false,
        "to": Object {
          "opacity": 1,
        },
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
